src/main/java/com/hhplus/commerce/spring/api/service/order/OrderService.java

```diff
@@ -0,0 +1,143 @@
+import com.hhplus.commerce.spring.api.service.order.request.OrderPaymentServiceRequest;
+import com.hhplus.commerce.spring.api.service.order.request.OrderServiceRequest;
+import com.hhplus.commerce.spring.api.service.payment.PaymentService;
+import com.hhplus.commerce.spring.model.entity.Order;
+import com.hhplus.commerce.spring.model.entity.OrderItem;
+import com.hhplus.commerce.spring.model.entity.Payment;
+import com.hhplus.commerce.spring.model.entity.Product;
+import com.hhplus.commerce.spring.model.entity.User;
+import com.hhplus.commerce.spring.repository.OrderItemRepository;
+import com.hhplus.commerce.spring.repository.OrderRepository;
+import com.hhplus.commerce.spring.repository.PaymentRepository;
+import com.hhplus.commerce.spring.repository.ProductRepository;
+import com.hhplus.commerce.spring.repository.UserRepository;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Service
+@Slf4j
+public class OrderService {
+    private final PaymentService paymentService;
+    private final DataPlatformService dataPlatformService;
+
+    private final UserRepository userRepository;
+    private final ProductRepository productRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final PaymentRepository paymentRepository;
+
+    @Transactional
+    public List<OrderItem> orderPayment(OrderPaymentServiceRequest request) {
+
+        User user = userRepository.findById(request.getUserId())
+                                  .orElseThrow(() -> new IllegalArgumentException("존재하지 않은 사용자"));
+
+        List<Long> productKeys = extractStockProductNumbers(request.getOrders());
+
+        Map<Long, Product> productMap = createStockMapBy(productKeys);
+        Map<Long, OrderServiceRequest> orderMap = createCountingMapBy(request.getOrders());
+
+        int payMoney = deductStockQuantities(productKeys, productMap, orderMap);
+
+        if (user.getUserPoint() < payMoney) {
+            throw new IllegalArgumentException("사용자 잔액 부족");
+        }
+
+        Order order = Order.create(user);
+        Order saveOrder = orderRepository.save(order);
+
+        List<OrderItem> orderItems = createOrderItems(order, productKeys, productMap, orderMap);
+
+        boolean payResult = paymentService.pointPayment(user.getId(), user.getUserPoint(), payMoney);
+
+        Payment payment = Payment.create(order, payMoney);
+
+        if (!payResult) {
+            payment.paymentStatusFailed();
+            paymentRepository.save(payment);
+
+            saveOrder.orderStatusPaymentFail();
+            throw new IllegalArgumentException("결제 실패");
+        }
```

Comment:
```suggestion
        boolean payResult = paymentService.pointPayment(user.getId(), user.getUserPoint(), payMoney);
        if (!payResult) {
            payment.paymentStatusFailed();
            paymentRepository.save(payment);

            saveOrder.orderStatusPaymentFail();
            throw new IllegalArgumentException("결제 실패");
        }
        Payment payment = Payment.create(order, payMoney);
```
이 순서가 되면 코드가 더 잘 이해될 것 같네요.